addons/binding/org.openhab.binding.rfxcom.test/src/test/java/org/openhab/binding/rfxcom/internal/messages/RFXComInterfaceControlMessageTest.java

```diff
@@ -0,0 +1,211 @@
+        configuration.enableAC = false;
+        configuration.enableARC = false;
+        configuration.enableX10 = false;
+    }
+
+    private void testMessage(RFXComInterfaceMessage.TransceiverType transceiverType, RFXComBridgeConfiguration configuration, String data)
+            throws RFXComException {
+
+        assertArrayEquals(DatatypeConverter.parseHexBinary(data),
+            new RFXComInterfaceControlMessage(transceiverType, configuration).decodeMessage());
+    }
+
+    @Test
+    public void testUndecodedMessage() throws RFXComException {
+        configuration.enableUndecoded = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530080000000000000");
+    }
+
+    @Test
+    public void testImagintronixOpusMessage() throws RFXComException {
+        configuration.enableImagintronixOpus = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530040000000000000");
+    }
+
+    @Test
+    public void testByronSXMessage() throws RFXComException {
+        configuration.enableByronSX = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530020000000000000");
+    }
+
+    @Test
+    public void testRSLMessage() throws RFXComException {
+        configuration.enableRSL = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530010000000000000");
+    }
+
+    @Test
+    public void testLighting4Message() throws RFXComException {
+        configuration.enableLighting4 = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530008000000000000");
+    }
+
+    @Test
+    public void testFineOffsetVikingMessage() throws RFXComException {
+        configuration.enableFineOffsetViking = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530004000000000000");
+    }
+
+    @Test
+    public void testRubicsonMessage() throws RFXComException {
+        configuration.enableRubicson = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530002000000000000");
+    }
+
+    @Test
+    public void testAEBlyssMessage() throws RFXComException {
+        configuration.enableAEBlyss = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530001000000000000");
+    }
+
+    @Test
+    public void testBlindsT1T2T3T4Message() throws RFXComException {
+        configuration.enableBlindsT1T2T3T4 = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530000800000000000");
+    }
+
+    @Test
+    public void testBlindsT0Message() throws RFXComException {
+        configuration.enableBlindsT0 = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530000400000000000");
+    }
+
+    @Test
+    public void testProGuardMessage() throws RFXComException {
+        configuration.enableProGuard = true;
+        testMessage(_433_92MHZ_TRANSCEIVER, configuration, "0D00000203530000200000000000");
+    }
+
+//  @Test
```

Comment:
Yes it could be a seperate PR but it is also not completely misplaced here, because now people might accidentally disable it if they use this option, without any chance to enable it.